src/app/modules/deployment-configs/components/dc-tree/dc-form/dc-form.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable, Subscription } from 'rxjs';
+import { DeploymentConfigsFacade } from '@app/core/facades/deployment-configs.facade';
+import { DeploymentConfig } from '@app/core/data/types';
+import { Router } from '@angular/router';
+import { FormBuilder, FormGroup, Validators, FormArray } from '@angular/forms'
+import { tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'hs-dc-form',
+  templateUrl: './dc-form.component.html',
+  styleUrls: ['./dc-form.component.scss']
+})
+export class DcFormComponent implements OnInit {
+  formEmitter: Subscription;
+
+  configTemplate = {
+    name: ['', [
+      Validators.required
+    ]],
+    hpa: {
+      maxReplicas: null,
+      cpuUtilization: null,
+      minReplicas: null,
+    },
+    deployment: {replicaCount: null},
+    container: {
+      resources: {
+        requests: { memory: '', cpu: '' },
+        limits: { memory: '', cpu: '' },
+      },
+      env: []
+    },
+    pod: null
+  }
+  form: FormGroup;
+
+  constructor(
+    private readonly facade: DeploymentConfigsFacade,
+    private router: Router,
+    private fb: FormBuilder
+  ) { }
+
+  onSave() {
+    let fv = this.form.value;
+    this.facade.update({
+      ...fv,
+      name: fv.name[0],
+      container: {
+        ...fv.container,
+        env: fv.container.env.reduce((acc, cur) => {
+          acc[cur.key] = cur.value;
+          return acc;
+        }, {})
+      }
+    });
+    localStorage.clear();
+    this.router.navigate([`deployment_configs/${fv.name[0]}`]);
+  }
+
+  ngOnInit(): void {
+    let config;
+
+    if(localStorage.getItem("formValue") !== null) {
```

Comment:
Now your component depends on low-level module and his concrete implementation - localStorage. 
What if we want to use another storage?
You can create abstraction for storage your data, and use your concrete storage through dependency injection.